### PR TITLE
get original filename for box files

### LINKED ISSUE #417: Cloud provider fixes for box.com
### Descriptive summary

On production scholar.uc.edu, we had to implement a couple of fixes to get box.com imports working properly.  We likely need those same fixes in ucrate.

Review the fixes from scholar_uc below and implement them if they are still needed.

You can test box.com imports on scholar-dev.uc.edu

- [x] https://github.com/uclibs/scholar_uc/pull/1401
- [x] https://github.com/uclibs/scholar_uc/pull/1668
- [x] https://github.com/uclibs/scholar_uc/pull/1663

### PR DESCRIPTION
**Pull request by crowesn**
Fixes #417 

Get original filename for files uploaded via browse-everything/box.com. Here we need to reopen the job_io_wrapper to get the filename from http header for box files. Browse_everything defaults to pulling filename from the url bub box.com doesn't include.

Changes proposed in this pull request:
* Reopen model to add method to pull filename from header
* Add model spec

### CODE REVIEW

File: spec/models/job_io_wrapper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        expect(subject).to receive(:file_from_uploaded_file!).and_call_original
+        subject.read
+      end
+      it '#mime_type and #original_name draw from the uploaded_file' do
+        expect(subject.mime_type).to eq('image/jpeg')
+        expect(subject.original_name).to eq('image.jpg')
+      end
+    end
+
+    context 'with remote file' do
```
Comment (crowesn) [inline spec/models/job_io_wrapper_spec.rb]: This is the test to cover our override.

**Comment by crowesn** [comment]
@hortongn Wondering why I can't get the first travis instance to respect .rubocop.yml..

**Comment by hortongn** [comment]
I was noticing that too.  I'll take a look and see why Travis thinks there are rubocop violations.

**Comment by hortongn** [comment]
@crowesn So I looked at your commit and the merge commit.  Travis is having the weird rubocop errors on the merge commit and it looks like that has a different version of the .rubocop.yml file.

Commit: https://github.com/uclibs/ucrate/commit/68fa4fd2cb117e41ced8e861700fce2624252e65

Merge commit: https://github.com/uclibs/ucrate/commit/749369d809c8e5a276204d464d0e0a7930357717

You may want to try closing your PR and opening a new one to see if that fixes it.  Might be a glitch.

**Pull request closed without merging**